File: 1436. Destination City/solution.java
class Solution {
  public String destCity(List<List<String>> paths) {
    Set<String> pathsSet = new HashSet<>();
    for (int index = 0; index < paths.size(); index++){
      List<String> path = paths.get(index);
      pathsSet.add(path.get(0));
    }

    for (int index = 0; index < paths.size(); index++){
      List<String> path = paths.get(index);
      if (!pathsSet.contains(path.get(1))){
        return path.get(1);
      }
    }
    return "";
  }
}

File: 1436. Destination City/solution.py
class Solution(object):
  def destCity(self, paths):
    """
    :type paths: List[List[str]]
    :rtype: str
    """
    pathsSet = set()
    for path in paths:
      pathsSet.add(path[0])
      
    for path in paths:
      if path[1] not in pathsSet:
        return path[1]
    return ""